### ISSUE TITLE
3PS :: There is no exposed API to add custom classname on module wrapper level on JSX

### ISSUE THREAD

**Issue by fikrirasyid** [elegantthemes/create-divi-extension#218]
### Problem Description
There is no equivalent of PHP's `ET_Builder_Element->add_classname()` method on JSX for extension. Thus so far, there is no way to do this on JSX template:

```
<div class="et_pb_cta_0 custom_class_name">
```

### Steps To Reproduce
1.

**Comment by fikrirasyid** [comment]
Note: feel free to move (or notify me to move this issue) this to `create-divi-extension` if this issue is more fitting to be placed there. I've asked @Jronk about this and thought this over but the `The problem with this issue here is actually even us can't recreate official module using existing API...` part, imo, makes it risky to put it there 🤔

**Comment by lots0logs** [comment]
@fikrirasyid I think it's preferable to encourage adding custom classes on the elements inside our wrapper that they have direct control over instead of adding them to our wrappers, no?

**Comment by xxtesaxx** [comment]
As @lots0logs said, you shouldn't rely on Divi adding classes to your module. Instead simply return in your render function a top level wrapper to which you add your classes, e. g.:

return '<div class="custom_class_name">...</div>' in php or return (<div className="custom_class_name">...</div>) in react.

The only difference is that addressing your wrapper now is .et_pb_cta .custom_class_name instead of .et_pb_cta.custom_class_name

**Comment by lots0logs** [comment]
Discussed this a bit with @fikrirasyid on Slack and the concern is when extending an official module the PHP output has to be styled with ` .et_pb_cta.custom_class_name` while the JSX has to be styled with `.et_pb_cta .custom_class_name`. What we'd like to do is make everything consistent. We'll continue to discuss it and see if we can come up with a way to do that.

**Comment by xxtesaxx** [comment]
So the use case here would be to create a module which extends a base module and in render(), before returning super->render() you call ET_Builder_Element->add_classname() to add new classes to the wrapper automatically.

In react though there is no possibility to add additional classes before calling super.render() and thats the issue?

**Comment by lots0logs** [comment]
:indeed:

**Comment by fikrirasyid** [comment]
@xxtesaxx yeah, that is safer way in general but things will be better if we manage to keep it consistent.

**Comment by xxtesaxx** [comment]
Yeah, I think it would be a great idea to implement the same functions in the VB which we have available in php. Another example would be generating responsive CSS or adding attributes to sections and stuff like this.

**Comment by fikrirasyid** [comment]
@xxtesaxx related to generating responsive CSS, i've filed an issue at https://github.com/elegantthemes/create-divi-extension/issues/221

**Comment by ayubadiputra** [comment]
This is possible in D5 now.
